src/main/java/org/opensearch/security/support/ConfigHelper.java

```diff
@@ -67,27 +69,30 @@ public static void uploadFile(Client tc, String filepath, String index, CType cT
         final String configType = cType.toLCString();
         LOGGER.info("Will update '" + configType + "' with " + filepath + " and populate it with empty doc if file missing and populateEmptyIfFileMissing=" + populateEmptyIfFileMissing);
 
-        if (!populateEmptyIfFileMissing) {
-            ConfigHelper.fromYamlFile(filepath, cType, configVersion, 0, 0);
-        }
-
-        try (Reader reader = createFileOrStringReader(cType, configVersion, filepath, populateEmptyIfFileMissing)) {
-
-            final IndexRequest indexRequest = new IndexRequest(index)
-                    .id(configType)
-                    .opType(OpType.CREATE)
-                    .setRefreshPolicy(RefreshPolicy.IMMEDIATE)
-                    .source(configType, readXContent(reader, XContentType.YAML));
-            final String res = tc.index(indexRequest).actionGet().getId();
+        AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
+            if (!populateEmptyIfFileMissing) {
+                ConfigHelper.fromYamlFile(filepath, cType, configVersion, 0, 0);
+            }
 
-            if (!configType.equals(res)) {
-                throw new Exception("   FAIL: Configuration for '" + configType
-                        + "' failed for unknown reasons. Pls. consult logfile of opensearch");
+            try (Reader reader = createFileOrStringReader(cType, configVersion, filepath, populateEmptyIfFileMissing)) {
+
+                final IndexRequest indexRequest = new IndexRequest(index)
+                        .id(configType)
+                        .opType(OpType.CREATE)
+                        .setRefreshPolicy(RefreshPolicy.IMMEDIATE)
+                        .source(configType, readXContent(reader, XContentType.YAML));
+                final String res = tc.index(indexRequest).actionGet().getId();
+
+                if (!configType.equals(res)) {
+                    throw new Exception("   FAIL: Configuration for '" + configType
+                            + "' failed for unknown reasons. Pls. consult logfile of opensearch");
+                }
+                LOGGER.info("Doc with id '{}' and version {} is updated in {} index.", configType, configVersion, index);
+            } catch (VersionConflictEngineException versionConflictEngineException) {
+                LOGGER.info("Index {} already contains doc with id {}, skipping update.", index, configType);
             }
-            LOGGER.info("Doc with id '{}' and version {} is updated in {} index.", configType, configVersion, index);
-        } catch (VersionConflictEngineException versionConflictEngineException) {
-            LOGGER.info("Index {} already contains doc with id {}, skipping update.", index, configType);
-        }
+            return null;
```

Comment:
Don't think this return statement is necessary

Reply:
`AccessController` has only functions which returns something always, looks no so good but it is what it is.
Usually other plugins introduces wrappers around this class e.g. https://github.com/opensearch-project/OpenSearch/blob/main/plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/SocketAccess.java

Reply:
Interesting. I had an understanding that void types do not require returning nulls, but I found this example that does return null for Void type. https://github.com/opensearch-project/OpenSearch/blob/main/plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3RepositoryPlugin.java#L66-L77

I'm fine keeping this return statement as is.

Reply:
Yes I wanted to remove but compiler fails.

Reply:
thank you for verifying :D